ionc/ion_reader.c

```diff
@@ -2156,6 +2155,42 @@ iERR ion_reader_get_value_offset(hREADER hreader, POSITION *p_offset)
     iRETURN;
 }
 
+iERR ion_reader_get_value_position(hREADER hreader, int64_t *p_offset, int32_t *p_line, int32_t *p_col_offset) {
+    iENTER;
+    ION_READER *preader;
+    POSITION    offset;
+    int32_t     line = -1;
+    int32_t     col_offset = -1;
```

Comment:
This follows a pattern I saw repeated elsewhere in the same file, i.e. in the `ion_reader_get_value_offset` immediately above.  This seems to avoid modifying the output parameters at all unless the function is going to succeed.  That said, since these are just being passed down to `_ion_reader_text_get_value_position` which does the same thing, it's not really necessary in this case.  I'll do as you suggest.